src/Compilers/CSharp/Test/Emit3/FieldKeywordTests.cs

```diff
@@ -4752,6 +4752,1617 @@
+            // Initialize by assigning in constructor
+            var source = """
+                #nullable enable
+
+                class C
+                {
+                    public static string Prop { get => field; set => field = value; }
+                    static C()
+                    {
+                        Prop = "a";
+                    }
+                }
+                """;
+
+            var comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics();
+        }
+
+        [Fact]
+        public void Nullability_06_Static_GetterOnly()
+        {
+            // Initialize by assigning to field in constructor (due to absence of setter)
+            var source = """
+                #nullable enable
+
+                class C
+                {
+                    public static string Prop { get => field; }
+                    static C()
+                    {
+                        Prop = "a";
+                    }
+                }
+                """;
+
+            var comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics();
+        }
+
+        [Fact]
+        public void Nullability_06_Static_NotInitialized()
+        {
+            var source = """
+                #nullable enable
+
+                class C
+                {
+                    public static string Prop { get => field; set => field = value; }
+                    static C() { }
+                }
+                """;
+
+            var comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics(
+                // (6,12): warning CS9264: Non-nullable property 'Prop' must contain a non-null value when exiting constructor. Consider adding the 'required' modifier, or declaring the property as nullable, or adding '[field: MaybeNull, AllowNull]' attributes.
+                //     static C() { }
+                Diagnostic(ErrorCode.WRN_UninitializedNonNullableBackingField, "C").WithArguments("property", "Prop").WithLocation(6, 12));
+        }
+
+        [Fact]
+        public void Nullability_06_Static_NotInitialized_GetterOnly()
+        {
+            var source = """
+                #nullable enable
+
+                class C
+                {
+                    public static string Prop { get => field; }
+                    static C() { }
+                }
+                """;
+
+            var comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics(
+                // (6,12): warning CS9264: Non-nullable property 'Prop' must contain a non-null value when exiting constructor. Consider adding the 'required' modifier, or declaring the property as nullable, or adding '[field: MaybeNull, AllowNull]' attributes.
+                //     static C() { }
+                Diagnostic(ErrorCode.WRN_UninitializedNonNullableBackingField, "C").WithArguments("property", "Prop").WithLocation(6, 12));
+        }
+
+        [Fact]
+        public void Nullability_07()
```

Comment:
>Nullability_07 [](http://example.com/codeflow?start=20&length=14)

Minor: It seems like `Nullability_08` covers this test. Similar comment for `_07_Getter_only`.

Reply:
The goal of this separation was to test the constructor exit checks separately from testing field flow state being reflected when reading the property.